src/aux-records/XpController.ts

```diff
@@ -0,0 +1,711 @@
+            };
+        if ('xpId' in config.receivingUserId) {
+            validationParams.authIds.push(config.receivingUserId.xpId);
+        } else if ('userId' in config.receivingUserId) {
+            validationParams.xpIds.push(config.receivingUserId.userId);
+        } else
+            return {
+                success: false,
+                errorCode: 'invalid_request',
+                errorMessage: 'The receiving user must have an xpId or userId.',
+            };
+
+        const batchParties = await this._validateUsers(validationParams);
+
+        if (!batchParties.success)
+            return {
+                success: false,
+                errorCode: 'user_not_found',
+                errorMessage: 'Some or all users were not found.',
+            };
+
+        let receivingUser = null;
+        let issuingUser = null;
+
+        if (
+            new Set(
+                batchParties.users.map((user) => {
+                    if (
+                        ('xpId' in config.receivingUserId &&
+                            user.id === config.receivingUserId.xpId) ||
+                        ('userId' in config.receivingUserId &&
+                            user.userId === config.receivingUserId.userId)
+                    ) {
+                        receivingUser = user;
+                    } else if (
+                        ('xpId' in config.issuingUserId &&
+                            user.id === config.issuingUserId.xpId) ||
+                        ('userId' in config.issuingUserId &&
+                            user.userId === config.issuingUserId.userId)
+                    ) {
+                        issuingUser = user;
+                    }
+                    return user.id;
+                })
+            ).size !== 2
+        )
+            return {
+                success: false,
+                errorCode: 'invalid_request',
+                errorMessage:
+                    'The issuing and receiving party cannot be the same user.',
+            };
+
+        return { success: true, receivingUser, issuingUser };
+    }
+
+    /**
+     * Get an Xp user's meta data (Xp meta associated with an auth user)
+     * Creates an Xp user for the auth user if one does not exist
+     */
+    async getXpUser(id: GetXpUserById): Promise<GetXpUserResult> {
+        return await tryScope(
+            async () => {
+                let user =
+                    (await ('userId' in id
+                        ? this._xpStore.getXpUserByAuthId(id.userId)
+                        : this._xpStore.getXpUserById(id.xpId))) ?? null;
+                if (!user && 'userId' in id) {
+                    const result = await this._createXpUser(id.userId);
+                    if (result.success) {
+                        user = result.user;
+                    } else return result;
+                }
+                return user
+                    ? { success: true, user }
```

Comment:
@LandonSiler Controllers should always try to be explicit about what they are returning so that we don't accidentally return more from the store than should be returned.  For example, instead of:
```js
return {
  success: true,
  user
};
```
I would prefer:
```js
return {
  success: true,
  user: {
    id: user.id,
    accountId: user.accountId,
    // etc...
  }
};
```
so we can see exactly what is being returned from the controller without having to consult the store.